File: c++/Neetcode_150/3_Stacks/14_valid_parentheses.cpp
//Valid Parentheses: Given a string s containing just the characters '(', ')', '{', '}', '[' and ']', determine if the input string is valid mathematically.

bool isValid(string s) {
        stack<char> st;
        for(auto i: s){
            if(i == '(' || i== '{' || i== '['){
                st.push(i);
            }
            else{
                if(st.empty()) return false;
                char open = st.top();
                if((open == '(' && i==')') || (open == '{' && i=='}') || (open == '[' && i==']')){
                    st.pop();
                }
                else return false;
            }
        }
        return st.empty();
    }

File: js/Neetcode_150/3_Stacks/14_valid_parentheses.js
//Valid Parentheses: Given a string s containing just the characters '(', ')', '{', '}', '[' and ']', determine if the input string is valid mathematically.

var isValid = function(s) {
    let stack = []
    for(let c of s){
        if(c==='(' || c==='{'||c==='[') stack.push(c)
        else{
            if(!stack.length) return false;
            let x = stack[stack.length-1]
            if(
                (c === ')' && x === '(') || 
                (c === '}' && x === '{') || 
                (c === ']' && x === '[')
            ) stack.pop()
            else return false
        }
    }
    return stack.length === 0
};